localstack/services/kms/models.py

```diff
@@ -180,10 +170,10 @@ def __init__(self, key_spec: str):
 
         if key_spec.startswith("RSA"):
             key_size = RSA_CRYPTO_KEY_LENGTHS.get(key_spec)
-            self.key = rsa.generate_private_key(public_exponent=65537, key_size=key_size)
+            key = rsa.generate_private_key(public_exponent=65537, key_size=key_size)
         elif key_spec.startswith("ECC"):
             curve = ECC_CURVES.get(key_spec)
-            self.key = ec.generate_private_key(curve)
```

Comment:
This here is the fix. Everything else is just refactoring to dissociate helper methods from the store definition.